### PR TITLE
moreutils: Add package

### PR DESCRIPTION
**Pull request by nikil**
Maintainer: me
Compile tested: ar71xx, x86_64
Run tested: x86_64

Description: additional Unix utilities

Signed-off-by: Nikil Mehta nikil.mehta@gmail.com

### CODE REVIEW

File: utils/moreutils/Makefile [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#
+# Copyright (C) 2006-2016 OpenWrt.org
```
Comment (diizzyy) [inline utils/moreutils/Makefile]: Please don't use OpenWrt.org copyright unless you are under contract.
https://www.lede-project.org/faq.html#who_holds_the_copyright_of_lede_code

File: utils/moreutils/Makefile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=moreutils
+PKG_VERSION:=0.59
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)_$(PKG_VERSION).orig.tar.gz
+PKG_SOURCE_URL:=http://http.debian.net/debian/pool/main/m/moreutils/
+PKG_MD5SUM:=22b26cebe53afae604f74f4135628217
```
Comment (diizzyy) [inline utils/moreutils/Makefile]: Please use SHA256 checksum

**Comment by nikil** [comment]
Thanks for your comments!  I've integrated your suggestions.

I didn't know that you could assign a SHA256 checksum to PKG_MD5SUM; it might be nice to have a separate variable like PKG_SHA256SUM for this to make it clearer for external contributors.

**Comment by diizzyy** [comment]
Indeed, I've brought that up for discussion.

**Comment by rotoroute** [comment]
I would like install this package in openwrt, it very good.  Many uses.

@diizzyy Why this not merged? Can you please merge?

**Comment by stintel** [comment]
@nikil can you change PKG_MD5SUM to PKG_HASH?

**Comment by diizzyy** [comment]
@rotoroute 
Probably little interest

**Comment by nikil** [comment]
@stintel I'm happy to make the change, but before I keep working on this, I'd like to know if this will ever be merged.

@diizzyy What is the bar for the amount of interest to merge a package?  Clearly I'd like this package in openwrt, and at least @rotoroute wants it.  Presumably @stintel is interested since he commented.  Is there a number of users that you are looking for before a package is accepted?

I'd like to be a regular contributor to openwrt (I've already added about 10 packages), but before I put the work in to make more contributions it would be helpful to have some clear guidelines about what the acceptance criteria is for packages.  And by clear guidelines, I mean something a contributor could know ahead of time before creating the package.

**Comment by stintel** [comment]
@nikil I asked to make that change so this PR can be merged. Looking at it again, I have a few more remarks, will review and comment inline.

File: utils/moreutils/Makefile [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=moreutils
+PKG_VERSION:=0.59
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)_$(PKG_VERSION).orig.tar.gz
+PKG_SOURCE_URL:=http://http.debian.net/debian/pool/main/m/moreutils/
+PKG_MD5SUM:=a48e11c3197bf79a7bfaa121423e64626e3381d9fedc91d606e9724ae498d1b4
```
Comment (stintel) [inline utils/moreutils/Makefile]: Change to PKG_HASH

File: utils/moreutils/Makefile [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+PKG_MAINTAINER:=Nikil Mehta <nikil.mehta@gmail.com>
+PKG_LICENSE:=GPL-2.0
+PKG_LICENSE_FILES:=COPYING
+
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION).orig
```
Comment (stintel) [inline utils/moreutils/Makefile]: Why is this needed? I think it should be dropped.

**Comment by nikil** [comment]
@stintel Great, thanks!  Looking forward to comments.

File: utils/moreutils/Makefile [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	- sponge: soak up standard input and write to a file
+	- ts: timestamp standard input
+	- vidir: edit a directory in your text editor
+	- vipe: insert a text editor into a pipe
+	- zrun: automatically uncompress arguments to command
+endef
+
+define Package/moreutils/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(PKG_BUILD_DIR)/{chronic,combine,errno,ifdata,ifne,isutf8,lckdo,mispipe,parallel,pee,sponge,ts,vidir,vipe,zrun} $(1)/usr/bin/
```
Comment (stintel) [inline utils/moreutils/Makefile]: Have you tried adding PKG_INSTALL:=1 on top and $(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin* ?

File: utils/moreutils/patches/001_disable-manuals.patch [hunk 6]
```diff
@@ -0,0 +1,11 @@
+--- a/Makefile
```
Comment (stintel) [inline utils/moreutils/patches/001_disable-manuals.patch]: I would just drop this patch completely. We generally don't do this for other packages either.

**Comment by stintel** [comment]
And while at it, bump to 0.60?

**Comment by stintel** [comment]
As for guidelines, https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md, https://wiki.openwrt.org/doc/devel/packages. When in doubt, check existing packages, ask on IRC, ML, forum, or just submit PR and await feedback :-)

File: utils/moreutils/Makefile (hunk 3 above)
Comment (nikil) [inline utils/moreutils/Makefile]: Done.

File: utils/moreutils/Makefile (hunk 4 above)
Comment (nikil) [inline utils/moreutils/Makefile]: This is needed because the package tarball unzips to moreutils-0.59.orig instead of moreutils-0.59.  Without this I can't get the package to build.

File: utils/moreutils/Makefile (hunk 5 above)
Comment (nikil) [inline utils/moreutils/Makefile]: PKG_INSTALL forces "make install" to be executed, and the "install" rule tries to install manpages.  As I mention below, I can't get the manpages to compile, so install will fail.

I think a better solution is to use PKG_INSTALL:=1 and to update my patch to also remove the installation of manpages.  I went ahead and did that - let me know what you think.

File: utils/moreutils/patches/001_disable-manuals.patch (hunk 6 above)
Comment (nikil) [inline utils/moreutils/patches/001_disable-manuals.patch]: This package won't compile without this patch - the default "all" make rule compiles manpages, and to compile manpages you need some obscure docbook commands (xsltproc?) that I can't seem to find.  There's no other rule in the unmodified makefile that will make all binaries but skip the manpages.

File: utils/moreutils/Makefile (hunk 4 above)
Comment (nikil) [inline utils/moreutils/Makefile]: I just bumped to 0.60, and it looks like this was fixed upstream.

**Comment by nikil** [comment]
@stintel Thanks for your comments!  Apologies for the delay, I had some issues with my build machine.

I've bumped the package to 0.60 and integrated your comments as best I could.  Please take a look.

File: utils/moreutils/patches/001_disable-manuals.patch (hunk 6 above)
Comment (diizzyy) [inline utils/moreutils/patches/001_disable-manuals.patch]: @stintel 
That is incorrect, we have several packages where docs (and other parts) are disabled using patching to reduce build time. Most of the time that "feature" has been requested.

Comment (stintel) [inline utils/moreutils/patches/001_disable-manuals.patch]: @nikil Thanks for the explanation.
@diizzyy At least provide proof for your claim. Some find and grep on both core and packages didn't show many packages where man-pages are disabled like this.

In general, I don't think it's worth the effort, unless it's required to fix breakage or avoid dependency bloat, so in this case it's fine.

**Comment by stintel** [comment]
1 minor nit: you probably should have added 2017 to the copyright line, but I'll accept as-is, you can add that on your next change (and I should have noticed it during the previous review).

**Comment by hnyman** [comment]
moreutils seems to fail to compile on both LEDE and Openwrt buildbots.
Looks like some kind of stripping problem.

http://buildbot.openwrt.org:8010/broken_packages/ar71xx/moreutils/compile.txt
http://downloads.lede-project.org/snapshots/faillogs/mips_24kc/packages/moreutils/compile.txt

Please fix the package.

**Comment by nikil** [comment]
Sorry about the delay.  I've submitted a PR with the fix: https://github.com/openwrt/packages/pull/4222
